data/campaigns/Heir_To_The_Throne/scenarios/13_The_Dwarven_Doors.cfg

```diff
@@ -160,7 +160,70 @@
         #
         {VARIABLE_OP true_entrance_location rand "1..2"}
 
-        # TODO: manually add snow detritus
+        [lua]
+            code=<<
+
+local helper = wesnoth.require "lua/helper.lua"
+
+-- The only parameter right now is the coverage (real 0.0 to 1.0, incl.)
+-- For prime-time use the filter terrain needs to come in and the convert
+-- table needs to be VERY complete. Or, rename this to morph_terrain and
+-- have the conversion table come in as an argument, too.
+
+local filter = {}
+filter.terrain = 'Ce,Ch,Chr,Co,Gd,Gd^Vc,Gd^Vh,Gd^Vo,Gll^Fp,Hhd,Hhd^Fp,Hhd^Vo,Ke,Khr,Ko,Md,Mm,Rd,Re,Re^Vo,Rr,Ww'
+local locations = wesnoth.get_locations(helper.shallow_parsed(filter))
+
+local coverage = wesnoth.get_variable("snowCoverage")
+if coverage == nil or coverage < 0 then
+    coverage = 0
+else
+    if coverage > 1 then
+        coverage = 1
+    end
+end
+local snowNeeded = (#locations * coverage) / 3
+-- Use 1/3rd of the coverage from Northern Winter. Spring is approaching. We need to watch
+-- this; if it adds too much snow, the player may not be able to make it to the exit in time.
+
+local convert = {}
```

Comment:
I think I'd prefer to see this table initialized all at once, like so:

```
local convert = {
    Ce = 'Cea',
    Ch = 'Cha',
}
```

etc.

If you prefer, you can write eg `['Ce'] = 'Cea'` instead; especially since this syntax is required for those terrain code keys that are not a valid identifiers.